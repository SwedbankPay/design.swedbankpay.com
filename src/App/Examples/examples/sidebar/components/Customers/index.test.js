import React from "react";
import { shallow } from "enzyme";

import {
    CustomersIntroduction,
    CustomersAdditionalInformation,
    COOverview,
    CDOrderHistory,
    CDInquiries,
    CDSummaryCharts
} from "./index";

describe("Examples: Sidebar - Customers", () => {
    describe("CustomersIntroduction", () => {
        it("is defined", () => {
            expect(CustomersIntroduction).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CustomersIntroduction />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CustomersAdditionalInformation", () => {
        it("is defined", () => {
            expect(CustomersAdditionalInformation).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CustomersAdditionalInformation />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("COOverview", () => {
        it("is defined", () => {
            expect(COOverview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<COOverview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CDOrderHistory", () => {
        it("is defined", () => {
            expect(CDOrderHistory).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CDOrderHistory test />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CDInquiries", () => {
        it("is defined", () => {
            expect(CDInquiries).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CDInquiries test />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("CDSummaryCharts", () => {
        it("is defined", () => {
            expect(CDSummaryCharts).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<CDSummaryCharts test />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
