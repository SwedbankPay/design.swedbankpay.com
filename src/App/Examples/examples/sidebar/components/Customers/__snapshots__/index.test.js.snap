// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples: Sidebar - Customers CDInquiries renders 1`] = `
<Fragment>
  <CustomersIntroduction />
  <h2
    id="customers-inquiries"
  >
    Customers inquiries
  </h2>
  <CustomersAdditionalInformation />
</Fragment>
`;

exports[`Examples: Sidebar - Customers CDOrderHistory renders 1`] = `
<Fragment>
  <CustomersIntroduction />
  <h2
    id="customers-order-history"
  >
    Customers order history
  </h2>
  <CustomersAdditionalInformation />
</Fragment>
`;

exports[`Examples: Sidebar - Customers CDSummaryCharts renders 1`] = `
<Fragment>
  <CustomersIntroduction />
  <h2
    id="customers-summary-charts"
  >
    Customers summary charts
  </h2>
  <CustomersAdditionalInformation />
</Fragment>
`;

exports[`Examples: Sidebar - Customers COOverview renders 1`] = `
<Fragment>
  <CustomersIntroduction />
  <h2
    id="customers-overview"
  >
    Customers overview
  </h2>
  <CustomersOverview
    customersList={
      Array [
        Object {
          "email": "sven.svensson@email.com",
          "firstName": "Sven",
          "id": "c1",
          "lastName": "Svensson",
          "location": "Visby",
          "phone": "+47 *** ** ***",
          "status": "default",
        },
        Object {
          "email": "anna.svensson@email.com",
          "firstName": "Anna",
          "id": "c2",
          "lastName": "Svensson",
          "location": "Stockholm",
          "phone": "+47 *** ** ***",
          "status": "success",
        },
        Object {
          "email": "ola.nordmann@email.com",
          "firstName": "Ola",
          "id": "c3",
          "lastName": "Nordmann",
          "location": "Oslo",
          "phone": "+47 *** ** ***",
          "status": "success",
        },
        Object {
          "email": "kari.nordmann@email.com",
          "firstName": "Kari",
          "id": "c4",
          "lastName": "Nordmann",
          "location": "Bergen",
          "phone": "+47 *** ** ***",
          "status": "warning",
        },
        Object {
          "email": "holger.danske@email.com",
          "firstName": "Holger",
          "id": "c5",
          "lastName": "Danske",
          "location": "Copenhagen",
          "phone": "+47 *** ** ***",
          "status": "danger",
        },
        Object {
          "email": "matti.meikalainen@email.com",
          "firstName": "Matti",
          "id": "c6",
          "lastName": "Meikäläinen",
          "location": "Lahti",
          "phone": "+47 *** ** ***",
          "status": "success",
        },
        Object {
          "email": "helma.egilsdottir@email.com",
          "firstName": "Helma",
          "id": "c7",
          "lastName": "Egilsdóttir",
          "location": "Reykjavík",
          "phone": "+47 *** ** ***",
          "status": "default",
        },
      ]
    }
    setCustomerIndex={[Function]}
    statusText={
      Object {
        "danger": "Missing",
        "default": "New",
        "success": "Delivered",
        "warning": "Late",
      }
    }
    test={true}
  />
  <CustomersAdditionalInformation />
</Fragment>
`;

exports[`Examples: Sidebar - Customers CustomersAdditionalInformation renders 1`] = `
<Fragment>
  <h2
    id="customers-additional-information"
  >
    Additional information
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque et mattis massa. Sed et gravida mi. Aliquam lobortis ligula nulla, vel consequat ligula auctor et. Vivamus placerat, odio et varius efficitur, neque leo condimentum mi, sed iaculis nisi arcu sit amet dui. Sed vehicula leo in purus lacinia, id auctor nulla scelerisque. Sed egestas sapien non enim pharetra sodales. Quisque a erat accumsan ligula efficitur vulputate blandit sed lorem. Nullam condimentum ultricies metus, et volutpat lacus tempor id. In tortor ante, interdum non convallis in, luctus non neque. Pellentesque in mauris enim.
  </p>
</Fragment>
`;

exports[`Examples: Sidebar - Customers CustomersIntroduction renders 1`] = `
<Fragment>
  <h2
    id="customers-introduction"
  >
    Introduction
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque et mattis massa. Sed et gravida mi. Aliquam lobortis ligula nulla, vel consequat ligula auctor et. Vivamus placerat, odio et varius efficitur, neque leo condimentum mi, sed iaculis nisi arcu sit amet dui. Sed vehicula leo in purus lacinia, id auctor nulla scelerisque. Sed egestas sapien non enim pharetra sodales. Quisque a erat accumsan ligula efficitur vulputate blandit sed lorem. Nullam condimentum ultricies metus, et volutpat lacus tempor id. In tortor ante, interdum non convallis in, luctus non neque. Pellentesque in mauris enim.
  </p>
</Fragment>
`;
